### PR TITLE
feat: Docs dark mode

### PR DESCRIPTION
**Pull request by felipecsl**
## Description

Not sure why but we had the theme switcher turned off. I just enabled the switch and set it to use the system preferred settings instead of defaulting to white.
It looks great on dark mode, I just added a white logo and it should be all set.
I also bumped Docusaurus to 2.4 since we were in a pretty old version. This version also allows you to change the theme via query string with `?docusaurus-theme=light|dark` other than the switcher on the top bar.

## Testing

Tested locally by switching back and forth between dark and light, played around with it, looks great.
image

### CODE REVIEW

**Review by leoromanovsky — APPROVED** [review]
Slick!

File: docusaurus.config.js [hunk 1]
```diff
@@ -105,7 +104,8 @@ async function createConfig () {
         // title: 'Eppo',
         logo: {
           alt: 'Eppo logo',
-          src: 'img/eppo_logo2024.svg'
+          src: 'img/eppo_logo2024.svg',
+          srcDark: 'img/eppo_logo2024_dark.svg'
```
Comment (aarsilv) [inline docusaurus.config.js:108]: neato!

File: docusaurus.config.js [hunk 2]
```diff
@@ -193,8 +193,8 @@ async function createConfig () {
       },
       colorMode: {
         defaultMode: 'light',
-        disableSwitch: true,
-        respectPrefersColorScheme: false
+        disableSwitch: false,
+        respectPrefersColorScheme: true
```
Comment (aarsilv) [inline docusaurus.config.js:197]: 📈

File: package.json [hunk 3]
```diff
@@ -14,9 +14,9 @@
     "write-heading-ids": "docusaurus write-heading-ids"
   },
   "dependencies": {
-    "@docusaurus/core": "^2.2.0",
-    "@docusaurus/plugin-sitemap": "^2.4.1",
-    "@docusaurus/preset-classic": "^2.2.0",
+    "@docusaurus/core": "^2.4.3",
+    "@docusaurus/plugin-sitemap": "^2.4.3",
+    "@docusaurus/preset-classic": "^2.4.3",
```
Comment (aarsilv) [inline package.json:19]: 📈

**Pull request merged**